cancermuts/datasources.py

```diff
@@ -1161,6 +1165,159 @@ def _get_revel_from_gc(self, mutation, gc):
             self.log.debug(f"Downloaded revel score: {revel_scores}")
             return [ DbnsfpRevel(source=self, score=r) for r in revel_scores ]
 
+class RevelLocal(DynamicSource, object):
+    @logger_init
+    def __init__(self, database_dir, database_files=None):
+        description = "Local REVEL score annotation using Ensembl transcript ID"
+        super(RevelLocal, self).__init__(name='RevelLocal', version='1.0', description=description)
+
+        self._database_dir = database_dir
+
+        if database_files is None:
+            self._database_files = {
+                'revel': os.path.join(self._database_dir, 'revel_with_transcript_ids')
+            }
+        else:
+            self._database_files = database_files
+
+        self._supported_metadata = {'revel_score': self._get_revel}
+
+    def add_metadata(self, sequence, md_type=['revel_score']):
+        if not sequence.uniprot_ac:
+            self.log.error("Sequence is missing UniProt accession; skipping.")
+            return
+
+        if type(md_type) is str:
+            md_types = [md_type]
+        else:
+            md_types = md_type
+
+        metadata_functions = []
+        for md in md_types:
+            try:
+                metadata_functions.append(self._supported_metadata[md])
+            except KeyError:
+                self.log.warning("RevelLocal doesn't support metadata type %s" % md)
+
+        try:
+            transcript_id = self._get_transcript_id(sequence)
+        except Exception as e:
+            self.log.error(f"Failed to retrieve transcript ID for {sequence}: {e}")
+            return
```

Comment:
I think this should happen in the `UniProt` object instead - as per the other identifiers and similarly to what happens for other data sources